proposals/gc/Post-MVP.md

```diff
@@ -403,6 +403,16 @@ However, there are a number of challenges:
 
 In general, the semantics and implementation of type parameters should be analogous to that of type imports. Ideally, in the presence of the [module linking proposal](https://github.com/WebAssembly/module-linking), it should even be possible to explain definitions with type parameters as shorthands for nested modules (well, at least for 2nd-class cases).
 
+### Fundamental Limitations
+
+Unfortunately, languages with generics exhibit one or both of the following features:
+* Expansive-recursive structures, such as `datatype 'a T = EMPTY | NODE of 'a * (('a T) T)` in ML
+* Impredicative bounded polymorphism, such as bounded generic methods in C#, or the bounded existential types used to describe (at the assembly level) Java/C# covariant arrays or Kotlin `in`/`out` type projections.
+
+Each of these features is undecidable on its own, and although the respective surface languages are decidable, their low-level encoding as structural types is not.
```

Comment:
Another false statement. For example, Harper/Stone's formalisation of Standard ML proves the opposite, and does not involve anything particularly fancy.

Reply:
Ah yes, these are not undecidable on their own as the surface languages are decidable. It is their respective extensions to the MVP that would be undecidable. For example, unlike the MVP, Harper/Stone's formalization has no subtyping (at this level), which it relies on for decidability. I pushed a qualification of the statement.